packages/experimental/Drawer/src/Drawer.types.ts

```diff
@@ -0,0 +1,195 @@
+import type { Animated, ColorValue, FlexAlignType, ModalProps, TouchableWithoutFeedbackProps, ViewProps } from 'react-native';
+
+import type { InteractionEvent, PressableFocusProps } from '@fluentui-react-native/interactive-hooks';
+
+export const DrawerName = 'Drawer';
+
+export type DrawerPositionType = 'left' | 'right' | 'bottom';
+
+export interface DrawerTokens {
+  /**
+   * The color of the backdrop.
+   * */
+  backdropColor?: ColorValue;
+
+  /**
+   * Width of the handle.
+   */
+  handleWidth?: number;
+
+  /**
+   * Height of the handle.
+   */
+  handleHeight?: number;
+
+  /**
+   * The background color of the handle.
+   * */
+  handleBackgroundColor?: ColorValue;
+
+  /**
+   * The corner radius of the handle.
+   * */
+  handleCornerRadius?: number;
+
+  /**
+   * The top margin of the handle.
+   *  */
+  handleMarginTop?: number;
+
+  /**
+   * The bottom margin of the handle.
+   * */
+
+  handleMarginBottom?: number;
+
+  /**
+   * The start margin of the handle.
+   * */
+
+  handleMarginStart?: number;
+
+  /**
+   * The end margin of the handle.
+   * */
+
+  handleMarginEnd?: number;
+
+  /**
+   * The flex alignment of the handle.
+   * */
+  handleAlignment?: FlexAlignType;
+
+  /**
+   * The background for drawer
+   * */
+
+  drawerBackgroundColor?: ColorValue;
+
+  /**
+   * The corner radius of the Drawer
+   * @default 0
+   * */
+  drawerCornerRadius?: number;
+
+  /**
+   * The elevation of the Drawer
+   * @default 5
+   * */
+  drawerElevation?: number;
+
+  /**
+   * The position of the Drawer
+   * @default 'left'
+   * */
+  position?: 'left' | 'right' | 'bottom';
+
+  /**
+   * The width of the Drawer
+   * Note: Only applicable when position is 'left' or 'right'
+   * @default '100%'
+   * */
+  width?: number | string;
+
+  /**
+   * The height of the Drawer
+   * Note: Only applicable when position is 'bottom'
+   * @default '40%'
+   * */
+  height?: number | string;
+
+  /**
+   * The shadow color of the Drawer
+   * @platform iOS
+   * */
+  shadowColor?: ColorValue;
+
+  /**
+   * The shadow offset of the Drawer
+   * @platform iOS
+   * */
+  shadowOffset?: { width: number; height: number };
+
+  /**
+   * The shadow opacity of the Drawer
+   * @platform iOS
+   * */
+  shadowOpacity?: number;
+
+  /**
+   * The shadow radius of the Drawer
+   * @platform iOS
+   * */
+
+  shadowRadius?: number;
+}
+
+export interface DrawerProps extends PressableFocusProps {
+  /*
+   ** An accessibility label for screen readers. Set on the text Drawer.
+   */
+  accessibilityLabel?: string;
+
+  /**
+   * Visibility of the Drawer
+   */
+  visible?: boolean;
+
+  /**
+   * Callback when the Drawer is closed
+   * */
+  onClose?: (e: InteractionEvent) => void;
+
+  /**
+   * Callback when the Drawer is opened
+   * */
+  onOpen?: (e: InteractionEvent) => void;
+
+  /**
+   * Whether the Drawer is open on mount
```

Comment:
How is this prop. intended to be used? If this new component is within another full screen RN experience - and drawer is expected to open on mount of some other root, would the developer need to set "`visible`" on mount or "`defaultOpen`" - and if latter, how do you plan to handle that..

Reply:
visible is the main criteria that will be used to define the state of Drawer - 'open' or 'close' .  

visible & defaultOpen will be working separately. 
defaultOpen depends upon visible : 

if visible is set of false - it becomes a control component and openDefault doesn't define anything.  

openDefault is applicable when the visible is true and screen is opened and Drawer visible = true.


defaultOpen is a property that decide the Drawer should open or not itself if the screen is loader where Drawer is used. 

Drawer will be not be tied to any sub rootview and will be always attached to topmost (zindex) view. 
if another Drawer comes it will be on top of previous Drawer (This is inlined with Native Drawer)

Reply:
Removed defaultOpen prop as Drawer seems be always controlled component which can be address via 'open' prop.